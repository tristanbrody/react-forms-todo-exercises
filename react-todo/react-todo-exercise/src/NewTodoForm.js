import React, { useState } from "react";

function NewTodoForm({ addTodo }) {
  const [formData, setFormData] = useState({});

  const handleChange = e => {
    const { name, value } = e.target;
    setFormData(data => ({
      ...data,
      [name]: value,
    }));
  };

  const handleSubmit = e => {
    e.preventDefault();
    addTodo(formData);
    setFormData("");
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="task">Add new to-do</label>
      <input
        type="text"
        name="task"
        value={formData.task}
        onChange={handleChange}
      ></input>
      <button>Add</button>
    </form>
  );
}

export default NewTodoForm;
