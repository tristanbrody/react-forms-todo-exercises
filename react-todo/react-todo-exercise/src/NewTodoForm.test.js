import { render, fireEvent } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

it("renders successfully", () => {
  render(<NewTodoForm />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});

it("fires function to change value for input", () => {
  // const handleChange = jest.fn();
  // const { asFragment, getByLabelText } = (<NewTodoForm />);
  // const form = getByLabelText('task');
  // fireEvent.change(form);
});
