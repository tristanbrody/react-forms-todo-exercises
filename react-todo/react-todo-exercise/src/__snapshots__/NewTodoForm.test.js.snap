// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="task"
    >
      Add new to-do
    </label>
    <input
      name="task"
      type="text"
      value=""
    />
    <button>
      Add
    </button>
  </form>
</DocumentFragment>
`;
